From 377564ee78bb475fab45b08151206bd183c9909b Mon Sep 17 00:00:00 2001
From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Wed, 4 Sep 2019 17:03:01 +0200
Subject: [PATCH] Add size of the pic_order_cnt related syntax elements to
 VASliceParameterBufferH264

Add the size in bits of the pic_order_cnt_lsb,
delta_pic_order_cnt_bottom, delta_pic_order_cnt[0], and
delta_pic_order_cnt[1] syntax elements.

The Hantro G1 hardware H.264 decoder requires this information.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 va/va.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/va/va.h b/va/va.h
index 70cf844..d990aa4 100755
--- a/va/va.h
+++ b/va/va.h
@@ -3153,9 +3153,16 @@ typedef struct _VASliceParameterBufferH264
     int16_t chroma_offset_l1[32][2];
     /** \brief Size in bits of the dec_ref_pic_marking() syntax element. */
     uint32_t dec_ref_pic_marking_bit_size;
+    /**
+     * \brief Size in bits of the pic_order_cnt related syntax elements.
+     *
+     * Size in bits of the pic_order_cnt_lsb, delta_pic_order_cnt_bottom,
+     * delta_pic_order_cnt[0], and delta_pic_order_cnt[1] syntax elements.
+     */
+    uint32_t pic_order_cnt_bit_size;
 
     /** \brief Reserved bytes for future use, must be zero */
-    uint32_t                va_reserved[3];
+    uint32_t                va_reserved[2];
 } VASliceParameterBufferH264;
 
 /****************************
